### ISSUE TITLE
iOS build fails with duplicate symbol errors

### ISSUE THREAD

**Issue by yoxallj** [googlevr/cardboard#52]
I am trying to upgrade an existing Cardboard project to use the new plugin.   Setting up the example project as described works perfectly.  In the real project however I get a bunch of errors when building for iOS, of the form:

duplicate symbol '_OBJC_IVAR_$_CardboardQRScanViewController._showingHUDMessage' in:

 .../Libraries/com.google.xr.cardboard/Runtime/iOS/**arm64/CardboardXrProvider.a**(qr_scan_view_controller_ced8551fbfbac309fbbe25b74b186b4f.o) 

.../Libraries/com.google.xr.cardboard/Runtime/iOS/**libsdk.a**(qr_scan_view_controller_ced8551fbfbac309fbbe25b74b186b4f.o)    

I have carefully checked that the project settings match between the two projects.

As these 2 files are both part of the cardboard package, do you have any ideas what might cause this?

The only thing the app does with VR is to play a 360 video, turning VR mode on and off using XRSettings.LoadDeviceByName("Cardboard");

**Comment by friggasaint** [comment]
I had similar problems.

If you use
XRSettings.LoadDeviceByName("Cardboard");
you are using the old plugin which is already deprecated for the App Store.

Make sure to only use only the new plug-in.
Therefore remove Google VR Android and Google VR iOS package from you package manager.
Make sure you have Google Cardboard XR Plugin for Unity left for the iOS build
This solved my issues.

**Comment by yoxallj** [comment]
> I had similar problems.
> 
> If you use
> XRSettings.LoadDeviceByName("Cardboard");
> you are using the old plugin which is already deprecated for the App Store.
> 
> Make sure to only use only the new plug-in.
> Therefore remove Google VR Android and Google VR iOS package from you package manager.
> Make sure you have Google Cardboard XR Plugin for Unity left for the iOS build
> This solved my issues.

Thanks for the response friggasaint.  This project was previously Unity 2018.4.13f1 and upgraded to use the new plugin, which may be relevant.

But that means I never installed the Google VR Android and Google VR iOS packages.  I have checked again, and they are definitely not installed.

I have also previously tried a build that commented out the call to XRSettings.LoadDeviceByName("Cardboard"), and it still failed in the same way (although now I think about it, I did not change the includes, so maybe still "using UnityEngine.XR" could be the problem? but that's where XRSettings is). 

Do you know how you would turn on VR for cardboard with the new plugin?  I could try that and see if it changes anything.

**Comment by Wang-YingGang** [comment]
I did the same as above, but the result is the same. Do you have other solutions?

**Comment by Wang-YingGang** [comment]
After I upgraded the Unity version to 2020.1.0a25, the problem disappeared.
I found that Unity 2020.1.0a25 is disabled in ProjectSettings->Player->XR Settings
So I think the conflict should be here

**Comment by Momeo** [comment]
2019.3.14 + iOS, same problem. After I delete libsdk.a in X-code settings I can build the project and run the app on my phone, but the 3dof rotation dont work at all.

**Comment by Wang-YingGang** [comment]
> 2019.3.14 + iOS, same problem. After I delete libsdk.a in X-code settings I can build the project and run the app on my phone, but the 3dof rotation dont work at all.

I tried the same operation as you to solve this problem, I changed my version to Unity2019.3.12f1, I did not encounter any problems

**Comment by yoxallj** [comment]
> After I upgraded the Unity version to 2020.1.0a25, the problem disappeared.
> I found that Unity 2020.1.0a25 is disabled in ProjectSettings->Player->XR Settings
> So I think the conflict should be here

Hi Wang.  I tried the upgrade you suggested, but unfortunately had a bunch of other errors which meant I couldn't get to the part of running an XCode build.

**Comment by yoxallj** [comment]
> 2019.3.14 + iOS, same problem. After I delete libsdk.a in X-code settings I can build the project and run the app on my phone, but the 3dof rotation dont work at all.

Thank you. I managed to get a successful build doing this as well.  To confirm whether I did the right thing, and for anyone else trying to repeat it, my steps were:
1. In XCode, select the target "Unity Framework"
2. Select the "Build Phases" tab
3. Expand the "Link Binary With Libraries" section
4. Select libsdka.a in the list,  then the  "-" at the  bottom of the list
